wallet/src/de/schildbach/wallet/ui/RestoreWalletFromSeedActivity.kt

```diff
@@ -0,0 +1,111 @@
+        return R.layout.activity_recover_wallet_from_seed
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setTitle(R.string.recover_wallet_title)
+    }
+
+    @SuppressLint("StringFormatInvalid")
+    fun onContinueClick(view: View) {
+        val seed = findViewById<EditText>(R.id.seed).text.trim()
+        if (seed.isEmpty()) {
+            return
+        }
+
+        val words = ArrayList(mutableListOf(*seed.split(' ').toTypedArray()))
+        try {
+            MnemonicCode.INSTANCE.check(words)
+            restoreWallet(WalletUtils.restoreWalletFromSeed(words, Constants.NETWORK_PARAMETERS))
+            log.info("successfully restored wallet from seed: {}", words.size)
+        } catch (x: IOException) {
+            val dialog = DialogBuilder.warn(this, R.string.import_export_keys_dialog_failure_title)
+            dialog.setMessage(getString(R.string.import_keys_dialog_failure, x.message))
+            dialog.setNeutralButton(R.string.button_dismiss, null)
+            dialog.show()
+            log.info("problem restoring wallet from seed: ", x)
+        } catch (x: MnemonicException) {
+            val dialog = DialogBuilder.warn(this, R.string.import_export_keys_dialog_failure_title)
+            dialog.setMessage(getString(R.string.import_keys_dialog_failure, x.message))
+            dialog.setNeutralButton(R.string.button_dismiss, null)
+            dialog.show()
+            log.info("problem restoring wallet from seed: ", x)
+        }
+
+    }
+
+    private fun restoreWallet(wallet: Wallet) {
+        walletApplication.replaceWallet(wallet)
+        getSharedPreferences(Constants.WALLET_LOCK_PREFS_NAME, Context.MODE_PRIVATE).edit().clear().apply()
+        configuration.disarmBackupReminder()
+
+        if (!wallet.hasKeyChain(Constants.BIP44_PATH)) {
+            if (wallet.isEncrypted) {
+                EncryptNewKeyChainDialogFragment.show(supportFragmentManager, Constants.BIP44_PATH)
+            } else {
+                // Upgrade the wallet now
+                wallet.addKeyChain(Constants.BIP44_PATH)
+                walletApplication.saveWallet()
+                // Tell the user that the wallet is being upgraded (BIP44)
+                // and they will have to enter a PIN.
+                UpgradeWalletDisclaimerDialog.show(supportFragmentManager)
+            }
+        } else {
+            resetBlockchain()
+        }
+    }
+
+    private fun resetBlockchain() {
+        val dialog = DialogBuilder(this)
+        dialog.setTitle(R.string.restore_wallet_dialog_success)
+        dialog.setMessage(getString(R.string.restore_wallet_dialog_success_replay))
+        dialog.setPositiveButton(R.string.button_ok) { _, _ ->
+            walletApplication.resetBlockchain()
+            val intent = SetPinActivity.createIntent(walletApplication, R.string.set_pin_create_new_wallet)
+            SingleLiveEvent<Intent>().call(intent)
+        }
```

Comment:
Why is the user being prompted?

Reply:
Would be nice to give some confirmation of successful restore, it has some explanations what happened: _Your wallet was successfully restored. Its balance will be determined next. This will take some time._
I can remove it if you think it's redundant.

Reply:
Let's leave it in.